dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -113,6 +113,8 @@ def __init__(self,
 
         # We need data about the batches (like shape information),
         # so we need to run a single batch as part of setup to get that info
+        for p in self._pipes:
```

Comment:
I don't know if moving this here gives any value as we are calling self.next() anyway few lanes bellow.

Reply:
`next` is now doing:
- `ShareOutputs`
- copy outputs to FW
- `ReleaseOutputs`
- `Run`

The very first next will be missing a `Run` to get the outputs from. It has to be done once in the ctor.

Reply:
I was asking if moving prefetching from next to ctor improves anything. I was not asking about getting rid of it.

Reply:
It makes it not prefetch (schedule `queue_depth` runs of the pipeline) every time you call next. I guess that's improvement.

Reply:
We certainly needed to remove call to `_prefetch` as it was scheduling `queue_size` runs of the pipeline for every next, and we consumed only one.

In `next` we consume the output, so we need to schedule a new `_run` after that.  We also need to start full run for the first iteration (with prefetching that will fill the queues). As we need to call this first `run` that causes the prefetching once, we do it in constructor.